testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -942,61 +926,61 @@ def test_copy(self, func, ts):
         ts = u.trajectory.ts
         func(self, self.name, ts)
 
-    def test_copy_slice(self):
-        for p, v, f in itertools.product([True, False], repeat=3):
-            if not any([p, v, f]):
-                continue
-            ts = self._from_coords(p, v, f)
-            yield self._check_copy, self.name, ts
-            yield self._check_independent, self.name, ts
-            yield self._check_copy_slice_indices, self.name, ts
-            yield self._check_copy_slice_slice, self.name, ts
-
-    def _check_bad_slice(self, p, v, f):
-        ts = self._from_coords(p, v, f)
+    @pytest.fixture(params=filter(any,
+                                  itertools.product([True, False], repeat=3)))
+    def some_ts(self, request):
+        p, v, f = request.param
+        return self._from_coords(p, v, f)
+
+    def test_copy_slice_no_yield(self, some_ts):
+        ts = some_ts
+        self._check_copy(self.name, ts)
```

Comment:
I'd rather all these method calls became separate tests

Reply:
Why? Just for style reasons?

Reply:
What if one of the `_check` calls alters `ts`?